src/Sarif.Sarifer/AnalyzeProjectCommand.cs

```diff
@@ -41,6 +40,10 @@ private void MenuCommandCallback(object caller, EventArgs args)
         {
             ThreadHelper.ThrowIfNotOnUIThread();
 
+            // always cancel before start
+            this.cancellationTokenSource?.Cancel();
+            this.cancellationTokenSource = new CancellationTokenSource();
+
```

Comment:
@michaelcfanning Suppose you start analyses on two different projects (with two separate gestures). If you cancel one of them (on the day we provide a "Cancel" gesture), do you want to cancel both of them?

If not, we need a mapping from project files to cancellation token sources. We can't get by with just one source.

Reply:
I don't see the value in having the ability to initiate two asynchronous analyses. You should think of this like a build gesture, you can build the SLN, or you can build a single project. Whatever you choose, you can't initiate another build operation until the first one is completed or canceled. Our feature should be the same. 


---
In reply to: [533006458](https://github.com/microsoft/sarif-visualstudio-extension/pull/302#discussion_r533006458) [](ancestors = 533006458)

Reply:
Based on this, this PR is still valid. If we press `analyze` in any project and press it again, it would cancel the first one and maintain the second. In the other hand, still checking how we will be able to cancel everything when we do this for sln level or when we close the sln

---
In reply to: [533014515](https://github.com/microsoft/sarif-visualstudio-extension/pull/302#discussion_r533014515) [](ancestors = 533014515,533006458)